src/v/cloud_storage/tx_range_manifest.cc

```diff
@@ -47,10 +243,34 @@ ss::future<> tx_range_manifest::update(ss::input_stream<char> is) {
     });
     iobuf_istreambuf ibuf(result);
     std::istream stream(&ibuf);
-    Document m;
     IStreamWrapper wrapper(stream);
-    m.ParseStream(wrapper);
-    do_update(m);
+
+    switch (_deserializer) {
```

Comment:
why keeping both implementations? I think it's fine to remove the old one completely.